index.test-d.ts

```diff
@@ -5,3 +5,9 @@ const clone = rfdc();
 
 expectType<number>(clone(5));
 expectType<{ lorem: string }>(clone({ lorem: "ipsum" }));
+
+const cloneHandlers = rfdc({
+  constructorHandlers: [
+    [RegExp, (o) => new RegExp(o)],
```

Comment:
I don't know if anyone else has better Type-Fu than me, but I can't seem to get the compiler to infer `o` as type `RegExp`.

<img width="320" alt="Screenshot 2024-06-12 at 11 17 38" src="https://github.com/davidmarkclements/rfdc/assets/12036746/bf8bdf9a-7a15-4c81-950b-ab1e445edfab">

I _think_ this is a [limitation of TypeScript](https://github.com/microsoft/TypeScript/issues/14466)